browser/src/communication/federated/federated_client.ts

```diff
@@ -119,13 +123,34 @@ export class FederatedClient extends Client {
     }
   }
 
+  private addDifferentialPrivacy (currentRoundWeights: Weights): Weights {
+    const weightsDiff = List(currentRoundWeights)
+      .zip(List(this.previousRoundWeights))
+      .map(([w1, w2]) => w1.add(-w2))
+
+    const normBound = weightsDiff.map((w) => w.norm().dataSync()[0]).max()
+
+    const noisyWeightsDiff = weightsDiff.map((w) => {
+      const clipped = w.div(Math.max(1, w.norm().dataSync()[0] / normBound))
+      const noise = tf.randomNormal(w.shape, 0, normBound * normBound)
```

Comment:
rename normBound -> norm ?  also make a configurable scalar parameter (from taskconfig) in front of this one? if zero, there will be no noise added (actually pack this one into an if-statement, as it's probably quite slow if there are millions of weights. rand generators can take quite some time.  this way the DP can be switched on / off in the task config (also add a comment to explain this in the config and here)

Reply:
do we still want clipping when there is no noise added?

Reply:
these can be independent. the default in the task config would be `noiseLevel=0` and `clipping=infty`. so both deactivated. so if people want they can activate them separately (clipping gives robustness for example)